plugin-server/src/main/ingestion-queues/session-recording-v2/consumer.ts

```diff
@@ -183,9 +193,11 @@ export class SessionRecordingIngester {
             func: async () => {
                 const parsedMessages = await this.kafkaParser.parseBatch(messages)
                 const messagesWithTeam = await this.teamFilter.filterBatch(parsedMessages)
+                const messagesWithRetention = await this.retentionService.processBatch(messagesWithTeam)
```

Comment:
question: Do we need to fetch retention for each message? This looks like a lot of Redis queries on the hot path. Could we fetch the retention values for each sessions just before flushing to S3? We could batch those lookups and avoid so many round trips to Redis.

I'm worried that putting it on the hot path will significantly slow down message processing.